PokeStar/PokeStar/DataModels/Raid.cs

```diff
@@ -1,131 +1,120 @@
       public DateTime CreatedAt { get; private set; }
+      private Dictionary<SocketGuildUser, int> Invite { get; set; }
 
       public Raid(short tier, string time, string location, string boss = null)
       {
          Tier = tier;
          Time = time;
          Location = location;
-         PlayerCount = 0;
-         HereCount = 0;
          SetBoss(boss);
-         Attending = new Dictionary<SocketGuildUser, int>();
-         Here = new Dictionary<SocketGuildUser, int>();
+         Groups = new List<RaidGroup>
+         {
+            new RaidGroup()
+         };
          Invite = new Dictionary<SocketGuildUser, int>();
+         RaidBossSelections = new List<string>();
          CreatedAt = DateTime.Now;
       }
 
-      public void PlayerAdd(SocketGuildUser player, int partySize, bool isInvite = false)
+      public ImmutableDictionary<SocketGuildUser, int> GetReadonlyInvite()
+      {
+         return Invite.ToImmutableDictionary(k => k.Key, v => v.Value);
+      }
+
+      public bool PlayerAdd(SocketGuildUser player, int partySize, SocketGuildUser user = null)
       {
-         if (Attending.ContainsKey(player))
+         int group;
+         if (user == null)
          {
-            int newPlayerCount = PlayerCount + (partySize - Attending[player]);
-            if (newPlayerCount <= playerLimit)
-            {
-               Attending[player] = partySize;
-               PlayerCount = newPlayerCount;
-            }
+            group = IsInRaid(player);
+            if (group == -1)
+               group = FindSmallestGroup();
+            Groups.ElementAt(group).Add(player, partySize);
          }
-         else if (Here.ContainsKey(player))
+         else // is invite
          {
-            int newPlayerCount = PlayerCount + (partySize - Here[player]);
-            if (newPlayerCount <= playerLimit)
+            group = IsInRaid(user);
+            if (group != -1)
             {
-               HereCount += partySize - Here[player];
-               PlayerCount = newPlayerCount;
-               Here[player] = partySize;
+               Groups.ElementAt(group).Invite(player, user);
+               return false;
             }
          }
-         else if (Invite.ContainsKey(player))
+         var newGroup = Groups.ElementAt(group).SplitGroup();
+         if (newGroup != null)
+            Groups.Add(newGroup);
+         CheckMergeGroups();
+         return true;
+      }
+
+      public void RemovePlayer(SocketGuildUser player)
+      {
+         if (Invite.ContainsKey(player))
          {
-            if (isInvite)
-            {
-               int newPlayerCount = PlayerCount + partySize;
-               if (newPlayerCount <= 20)
-               {
-                  Attending.Add(player, partySize);
-                  PlayerCount = newPlayerCount;
-               }
-            }
+            Invite.Remove(player);
+            return;
          }
-         else
+         foreach (var group in Groups)
```

Comment:
var -> PlayerGroup

Reply:
Will create issue for documentation branch that is to update all usages of var